### PR TITLE
Draft updated to address the comments from Sergio and Jeff

### LINKED ISSUE #47: Sergio comments on draft-yg3bp-ccamp-network-inventory-yang-01
[draft-yg3bp-ccamp-network-inventory-yang-01-sb-comments.txt](https://github.com/italobusi/ietf-network-inventory/files/9181277/draft-yg3bp-ccamp-network-inventory-yang-01-sb-comments.txt)

### LINKED ISSUE #48: Jeff comments on draft-yg3bp-ccamp-network-inventory-yang-01
[draft-yg3bp-ccamp-network-inventory-yang-01-sb-comments-jfb comments.docx](https://github.com/italobusi/ietf-network-inventory/files/9181278/draft-yg3bp-ccamp-network-inventory-yang-01-sb-comments-jfb.comments.docx)

### PR DESCRIPTION
**Pull request by italobusi**
Draft updated based on the 2022-08-04 call to address the comments from Sergio and Jeff: fix #47 and fix #48

Co-Authored-By: YuChaode <25008042+YuChaode@users.noreply.github.com>

### CODE REVIEW

**Comment by italobusi** [comment]
**[2022-09-07 Network Inventory weekly call](https://github.com/italobusi/ietf-network-inventory/blob/main/minutes/minutes-2022-09-07.md)**

> Pending comments:
> - a comment from Jeff about the common attributes for the network elements;

Jeff： provide some suggestion sentences.

> - how to improve the last sentence in section 2.2

Jeff： provide some suggestion sentences.
Chaode: provide some more description about the generic model at the beginning of draft. This generic refers to not only the generic components modeling idea, but also the technology-agnostic objective.

> - what to do with section 2.2 since the scalability issues are outside the scope of this draft.

Chaode: I suggest to keep it, and if there are some other solution provided by protocol level, we can provide some reference information here.

**Comment by italobusi** [comment]
**[2022-09-14 Network Inventory weekly call](https://github.com/italobusi/ietf-network-inventory/blob/main/minutes/minutes-2022-09-14.md)**

- [ ] @jbouqui153  to send text proposal
- [ ] @YuChaode  to update the draft on draft PR #49 based on last meeting and Jeff's input

Afterwards, the PR #49 can be opened for review

**Comment by jbouqui153** [comment]
Regarding the last sentence of section 2.2 I propose to rephrase it as follows if everybody is fine: "We consider that this generic data model, enabling a flexible and backward compatible approach for other technologies, represents the main scope of this draft. Solution description to efficiency/scalability limitations mentioned above is considered as out-of-scope."

**Comment by italobusi** [comment]
I am fine with the proposed text changes

File: draft-yg3bp-ccamp-network-inventory-yang.md [hunk 1]
```diff
@@ -507,30 +504,29 @@ We consider that some attributes defined in {{?RFC8348}} for components are also
             ...................................
 ~~~~
 
-Note: the attributes of network element are still under discussing.
+Note: Not all the attributes defined in {{?RFC8348}} are applicable for network element. And there could also be some missing attributes which are not recognized by {{?RFC8348}}. We will do some more extensions after the missing attributes are fully discussed.
```
Comment (aguoietf) [inline draft-yg3bp-ccamp-network-inventory-yang.md:507]: More extensions could be introduced in later revisions after the missing attributes are fully discussed.

File: draft-yg3bp-ccamp-network-inventory-yang.md [hunk 2]
```diff
@@ -507,30 +504,29 @@ We consider that some attributes defined in {{?RFC8348}} for components are also
...
 
+In case there are some other special types of inventory objects that could be used in other technologies and have not been considered by us, we would like to provide a generic model. If we define the inventory objects directly and give them fix hierarchical relationships in YANG model, once there is a new type of inventory object that needs to be introduced into the model, we need to break down our YANG model and insert the new one, this is not a backward compatible change and therefore is not an acceptable approach for implementation. With a generic model, we only need to augment a new component class and extend some specific attributes for this new inventory component class, which is more flexible. We consider that this generic data model, enabling a flexible and backward compatible approach for other technologies, represents the main scope of this draft. Solution description to efficiency/scalability limitations mentioned above is considered as out-of-scope.
```
Comment (aguoietf) [inline draft-yg3bp-ccamp-network-inventory-yang.md:517]: The model proposed by this draft is designed to be as generic as possible so to cover future special types of inventory objects that could be used in other technologies, that have not been identified yet. If the inventory objects were to be defined directly with fixed hierarchical relationships in YANG model, this new type of inventory objects needs to be manually defined, which is not a backward compatible change and therefore is not an acceptable approach for implementation. With a generic model, it is only needed to augment a new component class and extend some specific attributes for this new inventory component class, which is more flexible. We consider that this generic data model, enabling a flexible and backward compatible approach for other technologies, represents the main scope of this draft. Solution description to efficiency/scalability limitations mentioned above is considered as out-of-scope.

**Review by aguoietf — APPROVED** [review]

**Pull request merged**